### PR TITLE
SMP-1286 Integrate monitoring for harness services in helm-common repo

### PR DESCRIPTION
**Pull request by sarthak-harnessio**
- Added _monitoring.tpl 
- Bumped up chart version

### CODE REVIEW

File: charts/harness-common-1.0.12.tgz
Comment (kapilgarg1996) [inline charts/harness-common-1.0.12.tgz:1]: please delete this file, its not required here

File: src/common/templates/_monitoring.tpl [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+prometheus.io/path: {{ $path | quote }}
+prometheus.io/port: {{ default "8889" $port | quote }}
+prometheus.io/scrape: {{ $enabled | quote}}
+{{- end }}
+{{- end -}}
+
+{{/* Port to be added in deployment.yaml for java based applications
+{{ include "harnesscommon.monitoring.deploymentPort" . }}
+*/}}
+{{- define "harnesscommon.monitoring.deploymentPort" -}}
```
Comment (kapilgarg1996) [inline src/common/templates/_monitoring.tpl:34]:
```suggestion
{{- define "harnesscommon.monitoring.containerPort" -}}
```

File: src/common/templates/_monitoring.tpl [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+{{- $enabled := (pluck "enabled" $monitoring) | first }}
+{{- if $enabled }}
+prometheus.io/path: {{ $path | quote }}
+prometheus.io/port: {{ default "8889" $port | quote }}
+prometheus.io/scrape: {{ $enabled | quote}}
+{{- end }}
+{{- end -}}
+
+{{/* Port to be added in deployment.yaml for java based applications
+{{ include "harnesscommon.monitoring.deploymentPort" . }}
```
Comment (kapilgarg1996) [inline src/common/templates/_monitoring.tpl:32]:
```suggestion
{{ include "harnesscommon.monitoring.containerPort" . }}
```

File: charts/harness-common-1.0.12.tgz
Comment (sarthak-harnessio) [reply charts/harness-common-1.0.12.tgz:1]: sure, resolved

File: src/common/Chart.yaml [hunk 3]
```diff
@@ -15,7 +15,7 @@ type: library
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 1.0.12
+version: 1.0.13
```
Comment (kapilgarg1996) [inline src/common/Chart.yaml:18]: update the version to 1.0.14

**Review by kapilgarg1996 — APPROVED** [review]

**Pull request merged**